bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -1012,6 +1012,12 @@ public void scanEntryLog(long entryLogId, EntryLogScanner scanner) throws IOExce
             while (true) {
                 // Check if we've finished reading the entry log file.
                 if (pos >= bc.size()) {
+                    LOG.warn("read position illegal, ledgersMap maybe error. position {} entrylog size {} entrylog {}",
+                            pos, bc.size(), entryLogId);
+                    break;
+                }
+                if (pos < 0) {
```

Comment:
I'm curious about how `pos < 0` happens

Reply:
good catch, it looks like it is not possible

Reply:
at least throw in this case?

Reply:
This issue describes a scenario that causes pos to be negative.（#3604）

Reply:
> I'm curious about how `pos < 0` happens
This issue describes a scenario where pos is negative due to the disorder of entrylog files

Reply:
> good catch, it looks like it is not possible

This issue describes a scenario where pos is negative due to the disorder of entrylog files

Reply:
> I'm curious about how `pos < 0` happens

+1

Reply:
From the code logic, the `pos` won't be less than 0, and I'm not sure how `pos < 0` happens

Reply:
sorry，This problem exists in version 4.14, but not in the latest version.
<img width="1285" alt="image" src="https://user-images.githubusercontent.com/35036009/200124033-2643c421-02fb-403f-adfb-dc3bc7049aea.png">